gen_cert.py

```diff
@@ -85,14 +85,18 @@ def get_cert_date(
     - If using the "ROLLING" certs feature, use today's date
     - If all else fails use 'configured_date_parameter' for date
     """
-
+    utc_zone = get_timezone('UTC')
+    to_zone = get_timezone(timezone)
+    
     if calling_date_parameter:
-        date_value = format_datetime(calling_date_parameter, 'MMMM d, y', tzinfo=timezone, locale=locale)
+        date = strptime(calling_date_parameter, '%y-%m-%d')
     elif configured_date_parameter == "ROLLING":
-        date_value = format_datetime(datetime.datetime.today(), 'MMMM d, y', tzinfo=timezone, locale=locale)
+        date = datetime.datetime.today()
     else:
-        date_value = format_datetime(configured_date_parameter, 'MMMM d, y', tzinfo=timezone, locale=locale)
+        date = strptime(configured_date_parameter, '%y-%m-%d')
 
+    date = date.replace(tzinfo=utc_zone)
+    date_value = format_date(date.astimezone(to_zone).date(), 'long', locale=locale)
```

Comment:
Let's make this a bit more clear by breaking it out...

```python
date = date.replace(tzinfo=utz_zone)
date = date.astimezone(to_zone)
date = date.date()
date_value = format_date(date, 'long', locale=locale)
```

This is easier to step through when debugging.